src/module/utils/model-collection.mjs

```diff
@@ -1,29 +1,58 @@
-/** @import DataModel from "@common/abstract/data.mjs" */
+/**
+ * @import DataModel from "@common/abstract/data.mjs";
+ * @import PseudoDocument from "../data/pseudo-documents/pseudo-document.mjs";
+ * @import Collection from "@common/utils/collection.mjs";
+ */
+
+import BaseAdvancement from "../data/pseudo-documents/advancements/base-advancement.mjs";
+import BasePowerRollEffect from "../data/pseudo-documents/power-roll-effects/base-power-roll-effect.mjs";
 
 /**
  * Specialized collection type for stored data models.
  * @param {Array<string, DataModel>} entries    Array containing the data models to store.
  * @template {DataModel} Model The model class contained by this collection.
- * @extends {foundry.utils.Collection<string, Model>}
+ * @extends {Collection<string, Model>}
  */
 export default class ModelCollection extends foundry.utils.Collection {
+  constructor(documentName, document, data) {
+    super();
+    const name = CONFIG[document.documentName].dataModels[document._source.type].metadata.embedded[documentName];
+    Object.defineProperties(this, {
+      name: { value: name, writable: false },
+      _source: { value: data, writable: false },
+      documentClass: { value: ModelCollection.documentClasses[documentName], writable: false },
+    });
```

Comment:
If you compare the `ActiveEffect` collection to the `Advancement` collection on eg a class item, you'll notice only few differences, main being the `name` being long-winded for the pseudo-collections (see my comment on the PR).

In dnd5e, they assign the collection name differently, eg `advancements` and `activities`. We don't have a collection name for `PowerRollEffect`s.